src/System Application/App/VS Code Integration/src/VsCodeIntegrationImpl.Codeunit.al

```diff
@@ -120,7 +120,12 @@ codeunit 8333 "VS Code Integration Impl."
     local procedure GetAppIdForObject(ObjectType: Option; ObjectId: Integer): Text
     var
         NavAppInstalledApp: Record "NAV App Installed App";
+        EmptyGuid: Guid;
     begin
+        // Objects in the system app range
+        if ObjectId >= 2000000000 then
+            exit(EmptyGuid);
+
         if AllObjWithCaption.ReadPermission() then begin
```

Comment:
unrelated to this pr but AllObjWithCaption should really be a local... and with it you should skip the .Reset().
Right now there's no issue because the variable is only used here and higher up as a dummy... but that won't last... someone will abuse it...